diff --git a/include/linear/any.h b/include/linear/any.h
index 8293ab9..1e9ef68 100644
--- a/include/linear/any.h
+++ b/include/linear/any.h
@@ -43,7 +43,7 @@ class any {
     copy_msgpack_object(a.object_, &object_, zone_);
     type = static_cast<linear::type::any::Type>(object_.type);
   }
-  any(const linear::type::nil&) : zone_(), object_(), type(NIL) {
+  any(const linear::type::nil_t&) : zone_(), object_(), type(NIL) {
   }
   any(const msgpack::object& o) : zone_() {
     copy_msgpack_object(o, &object_, zone_);
