diff --git a/include/linear/any.h b/include/linear/any.h
index b088145..268a7db 100644
--- a/include/linear/any.h
+++ b/include/linear/any.h
@@ -25,7 +25,7 @@ class LINEAR_EXTERN any {
   any(const any& a) : zone_() {
     copy_msgpack_object(a.object_, &object_, zone_);
   }
-  any(const linear::type::nil&) : zone_(), object_() {
+  any(const linear::type::nil_t&) : zone_(), object_() {
   }
   any(const msgpack::object& o) : zone_() {
     copy_msgpack_object(o, &object_, zone_);
