diff --git a/include/linear/nil.h b/include/linear/nil.h
index cc4acfd..f5027ac 100644
--- a/include/linear/nil.h
+++ b/include/linear/nil.h
@@ -1,6 +1,6 @@
 /**
  * @file nil.h
- * a class definition of linear::type::nil
+ * a class definition of linear::type::nil_t
  */
 
 #ifndef LINEAR_TYPE_NIL_H_
@@ -17,18 +17,18 @@ namespace linear {
 namespace type {
 
 /**
- * @class nil nil.h "linear/nil.h"
+ * @class nil_t nil.h "linear/nil.h"
  * represent nil object
  * @warning
  * on iOS environment, you need to use nil_t() instead of nil().
  * (patched by linear-objc/bootstrap)
  */
-class nil {
+class nil_t {
  public:
   /// @cond hidden
-  nil() {
+  nil_t() {
   }
-  ~nil() {
+  ~nil_t() {
   }
   template <typename Packer>
   void msgpack_pack(Packer& pk) const {
