diff --git a/include/linear/nil.h b/include/linear/nil.h
index 8b64627..2dba8c6 100644
--- a/include/linear/nil.h
+++ b/include/linear/nil.h
@@ -17,20 +17,20 @@ namespace linear {
 namespace type {
 
 /**
- * @class nil nil.h "linear/nil.h"
+ * @class nil_ nil.h "linear/nil.h"
  * represent nil object
  * @warning on iOS environment, you need to use nil_() instead of nil().
  */
-class nil {
+class nil_ {
  public:
   /// @cond hidden
-  nil() {
+  nil_() {
   }
-  ~nil() {
+  ~nil_() {
   }
   template <typename Packer>
   void msgpack_pack(Packer& pk) const {
-    pk.pack(msgpack::type::nil());
+    pk.pack(msgpack::type::nil_());
   }
   void msgpack_unpack(msgpack::object o) {
     if(o.type != msgpack::type::NIL) {
