diff --git a/include/linear/optional.h b/include/linear/optional.h
index 7165e1c..5bcc3e5 100644
--- a/include/linear/optional.h
+++ b/include/linear/optional.h
@@ -28,7 +28,7 @@ class optional {
  public:
   optional() : initialized_(false) {
   }
-  optional(linear::type::nil) : initialized_(false) {
+  optional(linear::type::nil_t) : initialized_(false) {
   }
   optional(const optional<T>& other) {
     initialized_ = other.initialized_;
@@ -41,7 +41,7 @@ class optional {
   ~optional() {
   }
 
-  optional& operator=(linear::type::nil) {
+  optional& operator=(linear::type::nil_t) {
     initialized_ = false;
     return *this;
   }
@@ -113,7 +113,7 @@ class optional {
     if (initialized_) {
       pk.pack(value_);
     } else {
-      pk.pack(linear::type::nil());
+      pk.pack(linear::type::nil_t());
     }
   }
   void msgpack_unpack(msgpack::object o) {
@@ -268,19 +268,19 @@ bool operator<(const linear::type::optional<T>& lhs, const linear::type::optiona
 }
 
 template <class T>
-bool operator==(const linear::type::optional<T>& opt, linear::type::nil) {
+bool operator==(const linear::type::optional<T>& opt, linear::type::nil_t) {
   return !bool(opt);
 }
 template <class T>
-bool operator==(linear::type::nil, const linear::type::optional<T>&) {
+bool operator==(linear::type::nil_t, const linear::type::optional<T>&) {
   return false;
 }
 template <class T>
-bool operator<(const linear::type::optional<T>& opt, linear::type::nil) {
+bool operator<(const linear::type::optional<T>& opt, linear::type::nil_t) {
   return !bool(opt);
 }
 template <class T>
-bool operator<(linear::type::nil, const linear::type::optional<T>& opt) {
+bool operator<(linear::type::nil_t, const linear::type::optional<T>& opt) {
   return bool(opt);
 }
 
