diff --git a/include/linear/optional.h b/include/linear/optional.h
index c800f8a..2a12045 100644
--- a/include/linear/optional.h
+++ b/include/linear/optional.h
@@ -30,7 +30,7 @@ class LINEAR_EXTERN optional {
  public:
   optional() : initialized_(false) {
   }
-  optional(linear::type::nil) : initialized_(false) {
+  optional(linear::type::nil_) : initialized_(false) {
   }
   optional(const optional<T>& other) {
     initialized_ = other.initialized_;
@@ -43,7 +43,7 @@ class LINEAR_EXTERN optional {
   ~optional() {
   }
 
-  optional& operator=(linear::type::nil) {
+  optional& operator=(linear::type::nil_) {
     initialized_ = false;
     return *this;
   }
@@ -115,7 +115,7 @@ class LINEAR_EXTERN optional {
     if (initialized_) {
       pk.pack(value_);
     } else {
-      pk.pack(linear::type::nil());
+      pk.pack(linear::type::nil_());
     }
   }
   void msgpack_unpack(msgpack::object o) {
@@ -268,19 +268,19 @@ bool operator<(const linear::type::optional<T>& lhs, const linear::type::optiona
 }
 
 template <class T>
-bool operator==(const linear::type::optional<T>& opt, linear::type::nil) {
+bool operator==(const linear::type::optional<T>& opt, linear::type::nil_) {
   return !bool(opt);
 }
 template <class T>
-bool operator==(linear::type::nil, const linear::type::optional<T>&) {
+bool operator==(linear::type::nil_, const linear::type::optional<T>&) {
   return false;
 }
 template <class T>
-bool operator<(const linear::type::optional<T>& opt, linear::type::nil) {
+bool operator<(const linear::type::optional<T>& opt, linear::type::nil_) {
   return !bool(opt);
 }
 template <class T>
-bool operator<(linear::type::nil, const linear::type::optional<T>& opt) {
+bool operator<(linear::type::nil_, const linear::type::optional<T>& opt) {
   return bool(opt);
 }
 
